staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -402,8 +400,18 @@ redirectLoop:
 		// should be http://bar.com/c, not http://foo.com/c.
 		location, err = location.Parse(redirectStr)
 		if err != nil {
+			intermediateConn.Close()
 			return nil, nil, fmt.Errorf("malformed Location header: %v", err)
 		}
+
+		// Only follow redirects to the same host. Otherwise, propogate the redirect response back.
+		if validateRedirects && location.Hostname() != originalLocation.Hostname() {
+			break redirectLoop
```

Comment:
Shouldn't you be returning an error?

Reply:
No, this returns the redirect response back to the client. See the discussion on this comment thread: https://github.com/kubernetes/kubernetes/pull/66516#discussion_r210757111

Reply:
Do you think code calling this function is prepared to get a redirection response?

Reply:
(not to argue with @liggitt's comment-like in that thread, but... given what you told me in person I think it's probably right to go ahead and return the error, and just explicitly say that hostname to IP redirections (or vice versa) are not supported even if they happen to refer to the same thing.)

Reply:
If the CRI returned a different internal host address, then returning that to the client would be meaningless, and an error response would make more sense. However, if the pod returns a redirect  to an external service (trivial example: a pod that just returns a redirect response to a different external web address), then the expected behavior would be that the client can follow that redirect.

> Do you think code calling this function is prepared to get a redirection response?

 Looking at all the call sites, yes - it shouldn't be a problem. Also, this method only handles 302 redirects, so all other redirects were already being returned to the client.

Reply:
OK, sounds great :)